doc/tutorials/your_first_project/your_first_project.rst

```diff
@@ -0,0 +1,264 @@
+
+.. code-block:: bash
+
+  cd ~/ws_moveit2
+  source install/setup.bash
+
+Run your program and see the output.
+
+.. code-block:: bash
+
+  ros2 run hello_moveit hello_moveit
+
+The program should run and exit without error.
+
+2.2 Examine the code
+~~~~~~~~~~~~~~~~~~~~
+
+The headers included at the top are just some standard C++ headers and the header for ROS and MoveIt that we will use later.
+
+After that we have the normal call to initialize rclcpp and then we create our Node.
+
+.. code-block:: C++
+
+  auto const node = std::make_shared<rclcpp::Node>(
+    "hello_moveit",
+    rclcpp::NodeOptions().automatically_declare_parameters_from_overrides(true)
+  );
+
+The first argument is a string that ROS will use to make a unique node.
+The second is needed for MoveIt because of how we use ROS Parameters.
+
+Lastly we have the code to shutdown ROS.
+
+3 Plan and Execute using MoveGroupInterface
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In place of the comment that says "Next step goes here" add this code:
+
+.. code-block:: C++
+
+  // Create the MoveIt MoveGroup Interface
+  using moveit::planning_interface::MoveGroupInterface;
+  auto move_group_interface = MoveGroupInterface(node, "panda_arm");
+
+  // Set a target Pose
+  auto const target_pose = []{
+    geometry_msgs::msg::Pose msg;
+    msg.orientation.w = 1.0;
+    msg.position.x = 0.28;
+    msg.position.y = -0.2;
+    msg.position.z = 0.5;
+    return msg;
+  }();
+  move_group_interface.setPoseTarget(target_pose);
+
+  // Create a plan to that target pose
+  auto const [success, plan] = [&move_group_interface]{
+    moveit::planning_interface::MoveGroupInterface::Plan msg;
+    auto const ok = static_cast<bool>(move_group_interface.plan(msg));
+    return std::make_pair(ok, msg);
+  }();
+
+  // Execute the plan
+  if(success) {
+    move_group_interface.execute(plan);
+  } else {
+    RCLCPP_ERROR(logger, "Planing failed!");
+  }
+
+3.1 Build and Run
+~~~~~~~~~~~~~~~~~
+
+Just like before, we need to build the code before we can run it.
+
+In the workspace directory ``ws_moveit2`` run this command:
+
+.. code-block:: bash
+
+  colcon build --mixin debug
+
+After this succeeds, we need to re-use the demo launch file from the previous tutorial to start RViz and the MoveGroup node.
+In a separate terminal, source the workspace and then execute this:
+
+.. code-block:: bash
+
+  ros2 launch moveit2_tutorials demo.launch.py
+
+Then in the ``Displays`` window under ``MotionPlanning/Planning Request`` uncheck the box ``Query Goal State``.
+
+.. image:: rviz_1.png
+   :width: 300px
+
+In a third terminal source the workspace and run your program.
+
+.. code-block:: bash
+
+  ros2 run hello_moveit hello_moveit
+
+This should cause the robot in RViz to move and end up in this pose:
+
+.. image:: rviz_2.png
+   :width: 300px
+
+Note that if you ran the node ``hello_moveit`` without launching the demo launch file first it will wait for 10 seconds and then print this error and exit.
```

Comment:
```suggestion
Note that if you ran the node ``hello_moveit`` without launching the demo launch file first, it will wait for 10 seconds and then print this error and exit.
```